### PR TITLE
[CI] Remove actions that don't add value. 

### PR DESCRIPTION
**Pull request by corneil**
Remove actions that don't add value. 
Ensure documentation tests are executed.

### CODE REVIEW

File: .github/workflows/ci-pr.yml [hunk 1]
```diff
@@ -23,14 +16,7 @@ jobs:
...
       run: |
-        ./mvnw -B -s .github/settings.xml clean install
-    - name: Capture Test Results
-      if: failure()
-      uses: actions/upload-artifact@v3
-      with:
-        name: Unit Tests
-        path: '**/surefire-reports/*.xml'
-        retention-days: 3
+        ./mvnw -B -s .github/settings.xml -Pasciidoctordocs,restdocs clean install
```
Comment (onobc) [inline .github/workflows/ci-pr.yml:19]: This seems very specific to `asciidoctordocs,restdocs` -- will all the other tests execute? I did not know we even had these profiles. Where are they defined?

Comment (corneil) [reply .github/workflows/ci-pr.yml:19]: They have been in place to ensure the rest docs execute even when skipping tests but not during a normal build.

Comment (onobc) [reply .github/workflows/ci-pr.yml:19]: > will all the other tests execute?

Comment (corneil) [reply .github/workflows/ci-pr.yml:19]: I will simplify this to a single profile.

Comment (corneil) [reply .github/workflows/ci-pr.yml]: > > will all the other tests execute?

If you build with `./mvnw clean install -DskipTests -Pdocs` the docs will be generated.
If you build with `./mvnw clean install -DskipTests` the docs will be not generated and not tests executed.
If you build with `./mvnw clean install` the docs will be not generated and tests executed.

**Review by onobc — APPROVED** [review]
Thanks for the updates @corneil . The profiles are much simpler now. LGTM

**Pull request merged**